File: Week_02/G20190343020350/week_02/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        unordered_map<char, int> umap;

        for (int i = 0; i < s.size(); i++) {
            ++umap[s[i]];
            --umap[t[i]];
        }
        for (auto c:umap) {
            if (c.second != 0)
                return false;
        }
        return true;
    }
};

File: Week_02/G20190343020148/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        int[] counter=new int[26];
        for(int i=0;i<s.length();i++){
            counter[s.charAt(i)-'a']++;
        }
        for(int j=0;j<t.length();j++){
            counter[t.charAt(j)-'a']--;
            if(counter[t.charAt(j)-'a']<0){
                return false;
            }
        }
        return true;
    }
}